tailwind/ui/buttons/ButtonTwoLines.tsx

```diff
@@ -63,9 +63,20 @@ const ButtonTwoLines = ({
   )
 
   if (props.componentType === "link") {
+    const { buttonProps, ...rest } = props
     return (
-      <ButtonLink className={commonClassStyles} size={size} {...props}>
-        <ChildContent {...props} size={size} isIconLeft={isIconLeft} />
+      <ButtonLink
+        className={commonClassStyles}
+        // size={size}
```

Comment:
Did comment this out to skip a type error that was breaking the build. I see this as a minor issue for now since this component is not currently used. We can tackle the issue in a separate PR.

I also think that `<ButtonTwoLines componentType="link" />` is not working. Something we can look into later.

Reply:
With `componentType`, the type guarding is not working as expected. Need to re-evaluate the approach.

With `size`, it is housed in `buttonProps` when passing props to `ButtonLink`. Not going to try and make the approach any more strict, but can simply run a merge props of `buttonProps` with `size`.

```tsx
if (props.componentType === "link") {
  const { buttonProps, ...rest } = props

  const mergedButtonProps = merge(buttonProps, { size })
  return (
    <ButtonLink
      className={commonClassStyles}
      buttonProps={mergedButtonProps}
      {...rest}
    >
      <ChildContent
        {...rest}
        size={size}
        isSecondary={buttonProps?.isSecondary}
        isIconLeft={isIconLeft}
      />
    </ButtonLink>
  )
}
```
What do you think?

Reply:
I think we should follow the same approach we took with the original ButtonLink. In this case, I think we should return just a button to simplify everything. Then, in case you need a buttonlink, the user/dev could do the same we are doing with the buttonlink.

```ts
// always a button
<ButtonTwoLines size=".." variant="..." />

// in case you want a buttonlink => compose them with asChild (same as with ButtonLink)
<ButtonTwoLines asChild>
  <BaseLink> ...
</Button>
```

Reply:
@pettinarip Good with me! Getting too DRY here, and it's showing its cracks